### PR TITLE
Modernize dicts

### PR DESCRIPTION
**Pull request by encukou**
Add improvements from a few years of in-person courses.
There's been some back-and-forth in additional serialization formats and `bytes` in the JSON lesson.
The “multiple values for a single key” section was moved to a separate lesson – it's good to take a break and do some exercises before it.

The commits represent historical evolution, don't review them individually.

### CODE REVIEW

File: lessons/beginners/dict/index.md [hunk 1]
```diff
@@ -2,137 +2,147 @@
...
-str
-dict
-len
+Jablko
+Knoflík
+Myš
 ```
 
-Pokud chceš hodnoty, stačí použít metodu `values`:
+Kdybys chtěl{{a}} projít místo klíčů hoodnoty, použij metodu `values`,
```
Comment (Kobzol) [inline lessons/beginners/dict/index.md:124]:
```suggestion
Kdybys chtěl{{a}} projít místo klíčů hodnoty, použij metodu `values`,
```

File: lessons/beginners/dict/index.md [hunk 2]
```diff
@@ -141,52 +151,51 @@ len: délka
...
 ```python
 data = [(1, 'jedna'), (2, 'dva'), (3, 'tři')]
 nazvy_cisel = dict(data)
 ```
 
+nebo jiný iterovatelný objekt:
+
+```python
+data = enumerate(['nula', 'jedna', 'dva'])
+nazvy_cisel = dict(data)
```
Comment (Kobzol) [inline lessons/beginners/dict/index.md:196]: Tady by možná byla fajn ukázka, jaký slovník se z toho stane? Stejně tak u `nazvy_cisel` nahoře.

File: lessons/beginners/dict/index.md [hunk 3]
```diff
@@ -2,137 +2,147 @@
...
 ...     print('{}: {}'.format(klic, hodnota))
-str: řetězec
-dict: slovník
-len: délka
+Jablko: Apple
+Knoflík: Button
+Myš: Mouse
 ```
 
 > [note]
```
[the remark this answers is not in the document]
Comment (encukou) [reply lessons/beginners/dict/index.md:148]: No jo, asi máš pravdu...

File: lessons/projects/github-api/index.md [hunk 4]
```diff
@@ -104,9 +115,9 @@ Teď už je lépe vidět celý tvůj profil
 (možná včetně neveřejných informací – proto musíš svůj token
 udržovat v tajnosti).
 
-S profilem, který máš v proměnné `data`,
+S profilem který máš v proměnné `data`
```
[the remark this answers is not in the document]
Comment (janpipek) [reply lessons/projects/github-api/index.md:118]: Kde ti opravuju čárky, jsem si poměrně dost jist.

**Pull request merged**